analyses/ssgsea-hallmark/02-analyze-plot-ssgsea-res.R

```diff
@@ -0,0 +1,186 @@
+##########################################
+#Purpose: Code to analyze and plot ssGSEA Results
+#Author: Pichai Raman
+#Date: 9/23/2019
+##########################################
+
+#Call libraries
+library("tidyverse")
+library("pheatmap")
+library("viridis")
+library("optparse")
+
+# Declare command line options
+option_list <- list(
+  optparse::make_option(
+    c("-a", "--anova_pvalue"),
+    type = "double",
+    default = 0.01,
+    help = "ANOVA P-value threshold",
+  ),
+  optparse::make_option(
+    c("-t", "--tukey_pvalue"),
+    type = "double",
+    default = 0.05,
+    help = "Tukey HSD Adjusted P-value threshold"
+  ),
+  optparse::make_option(
+    c("-p", "--perc_keep"),
+    type = "double",
+    default = .25,
+    help = "Percentage of diseases a particular disease has a significantly higher score than for a specific pathway"
+  )
+)
+
+#Read the arguments passed
+opt_parser <- optparse::OptionParser(option_list = option_list)
+opt <- optparse::parse_args(opt_parser)
+
+#Pass arguments to variables
+aovParam <- opt$anova_pvalue
+padjParam <- opt$tukey_pvalue
+percKeep <- opt$perc_keep
+
+
+#Read in clinical data and pathway level data generated from 01-run-ssgsea.R
+clinData <- read.delim("../../data/pbta-histologies.tsv", stringsAsFactors=F)
+geneSetExpMat <- readRDS("../../scratch/GeneSetExpressionMatrix.RDS")
+
+#Choose Clinical Samples corresponding to samples in Gene Set Matrix
+#This is to makes sure that clinical data and the pathway DF have the same samples
+#and that they are ordered the same way
+clinData <- clinData[clinData[,"experimental_strategy"]=="RNA-Seq",]
+rownames(clinData) <- clinData[,"Kids_First_Biospecimen_ID"]
+clinData <- clinData[colnames(geneSetExpMat),]
+
+#Function to split pathway names from TukeyHSD into vector of 2 elements
+#i.e. Tukey HSD outputs Pathway1-Pathway2 and this will spitout c(Pathway1, Pathway2)
```

Comment:
If using `short_histology` this will be `Histology1-Histology2`, not `Pathway1-Pathway2`, correct?

Reply:
That's right, I think probably best to make it generic as you correctly state in another comment Clinical Variables may be  we may use something other than Histology so I've updated the comment accordingly. Will push changes shortly.